Add freeglut

jakirkham opened the pull request:
Attempts to build `freeglut` from source.

This is Linux only right now. Windows support is possible as people have and do build this on that platform.

It requires X11 support for Mac, which is not guaranteed for later versions of the OS. Mac OS 10.7 is the last to have XQuartz (Mac's X11) by default. On later OS versions it can be installed optionally, but we shouldn't have it as a hard dependency. See this issue ( https://github.com/dcnieho/FreeGLUT/issues/27 ) from a fork of FreeGLUT for a fading glimmer for hope. :cry:

conda-forge-admin commented:
Hi! This is the friendly conda-forge-admin automated user.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/freeglut```) and found it was in an excellent condition.

jakirkham commented:
After a bit of playing it looks like we have to `yum` install more stuff. Namely, `libXi-devel`, `mesa-libGLU-devel`, and possibly more.

jakirkham commented:
So, this works nicely on Linux now given the yum requirements are satisfied.

pelson commented on recipes/freeglut/yum_requirements.txt:
```diff
@@ -0,0 +1,2 @@
+libXi-devel
+mesa-libGLU-devel
```
We can remove this one now.

jakirkham replied:
Actually, we added `mesa-libGL-devel`. We still have to install `mesa-libGLU-devel` for GLU support.

pelson commented on scripts/run_docker_build.sh:
```diff
@@ -57,6 +57,9 @@ yum install -y libXext libXrender libSM tk libX11-devel mesa-libGL-devel
 rm -rf /conda-recipes/example
 
+# Installs anything from a \`yum_requirements.txt\` file in a recipe with \`yum\`.
+find conda-recipes -mindepth 2 -maxdepth 2 -type f -name "yum_requirements.txt" | xargs -n1 cat | xargs -r yum install -y
```
This line can be removed (it is already merged)

jakirkham replied:
Done.

msarahan commented:
**bump** - @jakirkham any time to address @pelson's comments?

jakirkham commented:
Not tonight. I think I'm going to call it.

jakirkham commented:
Though I have intentionally not dealt with these issues so that we don't merge it yet. The reason being we have not yet resolved yum requirements in conda-smithy. Once that is dealt with, I will clean this up. In all likelihood, this will probably be the first recipe to use yum requirements in a feedstock.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/freeglut```) and found it was in an excellent condition.

jakirkham commented:
Rebased so this doesn't have CI changes. Addressed relevant comments. Both `conda-smithy` and `staged-recipes` have support for yum requirements now so this should be mergeable.

jakirkham commented:
Sending it in. More feedback welcome in the feedstock. Thanks everyone. 😄
